### PR TITLE
HPCC-18380 WsSQL migration into platform

### PR DESCRIPTION
**Pull request by Michael-Gardner**
Signed-off-by: Michael Gardner <michael.gardner@lexisnexisrisk.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [x] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [x] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

### CODE REVIEW

**Comment by Michael-Gardner** [comment]
@rpastrana Please review, and lets talk about what we needs to change in the documentation now that this is going to be part of the platform?

**Comment by AttilaVamos** [comment]
@Michael-Gardner It seems this PR introduces some new dependency like ANTLR-3.4. Please update the wiki as well.

**Comment by AttilaVamos** [comment]
@Michael-Gardner Or... So we are doing a lots of (I don't like) on fly install/build stuff. Why we don't  download those ANTLR related jar files if they are missing?

File: esp/services/ws_sql/SQL2ECL/HPCCFile.hpp [hunk 1]
```diff
@@ -0,0 +1,389 @@
...
+#define HPCCFILE_HPP_
+
+#include "ws_sql.hpp"
+#include "SQLColumn.hpp"
+
+#include "ws_sql_esp.ipp"
+#include "hqlerror.hpp"
+#include "hqlexpr.hpp"
+
+/* undef SOCKET definitions to avoid collision in Antlrdefs.h*/
```
Comment (rpastrana) [inline esp/services/ws_sql/SQL2ECL/HPCCFile.hpp]: is this safe?

File: initfiles/componentfiles/configxml/cgencomplist_linux.xml [hunk 2]
```diff
@@ -113,6 +113,9 @@
 <Component name="ws_ecl" processName='EspService' schema='esp_service_wsecl2.xsd' deployable='no'>
     <File name="@temp/esp_service.xsl" method="esp_service_module"/>
   </Component>  
+  <Component name="ws_sql">
```
Comment (rpastrana) [inline initfiles/componentfiles/configxml/cgencomplist_linux.xml]: Jira claims you changed ~ 350 lines in initfiles/componentfiles/configxml/buildsetCC.xml.in, is that accurate? did you maybe reformat the file? in any case I wasn't able to view the diff.

File: cmake_modules/FindANTLR.cmake [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+option(ANTLR_VER "ANTLR runtime and buildtime version.")
+if(NOT ANTLR_VER)
+    set(ANTLR_VER 3.4)
+endif()
+
+option(ANTLR_BUILDTIME_DEP "ANTLR buildtime jar file name.")
+if(NOT ANTLR_BUILDTIME_DEP)
+    set(ANTLR_BUILDTIME_DEP "antlr-${ANTLR_VER}-complete")
+    message(STATUS "Option ANTLR_BUILDTIME_DEP not set, setting to: ${ANTLR_BUILDTIME_DEP}")
+    message(STATUS "The jar can be downloaded directly from:\n\thttps://github.com/antlr/website-antlr3/raw/gh-pages/download/antlr-3.4-complete.jar")
```
Comment (rpastrana) [inline cmake_modules/FindANTLR.cmake]: let's automatically download these... can that be a submodule? 
also, let's make a USE_ANTLR option, if false, don't build Wssql and proceed w/ the build

**Review by rpastrana — COMMENTED** [review]
@Michael-Gardner couple of comments

**Comment by rpastrana** [comment]
@Michael-Gardner a couple of comments/questions.

File: initfiles/componentfiles/configxml/cgencomplist_linux.xml (hunk 2 above)
Comment (Michael-Gardner) [reply initfiles/componentfiles/configxml/cgencomplist_linux.xml]: Yeah I fixed the indentation when I did that one.  I can revert it if you need.

**Comment by Michael-Gardner** [comment]
@rpastrana You can now turn the building of ws_sql service on or off by using -DWSSQL_SERVICE=OFF when you invoke cmake.

**Comment by rpastrana** [comment]
@Michael-Gardner looks good. We might need @xwang2713  to review the latest CMAKE changes.

**Review by xwang2713 — COMMENTED** [review]
CMake  Configure looks good

**Comment by rpastrana** [comment]
@Michael-Gardner this pr looks good to go.

**Comment by Michael-Gardner** [comment]
@richardkchapman PR is done, ready for merge.

**Pull request merged**